From 0dab3e0e59ac3692e2f95cb37ba17d5dfb3d1e5f Mon Sep 17 00:00:00 2001
From: "Michael S. Tsirkin" <mst@redhat.com>
Date: Tue, 5 Jan 2016 18:17:53 +0200
Subject: [PATCH] s390: drop smp_mb in vdso_init
Git-commit: 0dab3e0e59ac3692e2f95cb37ba17d5dfb3d1e5f
Patch-mainline: v4.5-rc1
References: bsc#969756

The initial s390 vdso code is heavily influenced by the powerpc version
which does have a smp_wmb in vdso_init right before the vdso_ready=1
assignment. s390 has no need for that.

Signed-off-by: Michael S. Tsirkin <mst@redhat.com>
Message-id: <1452010645-25380-1-git-send-email-mst@redhat.com>
Signed-off-by: Christian Borntraeger <borntraeger@de.ibm.com>
Signed-off-by: Martin Schwidefsky <schwidefsky@de.ibm.com>
Signed-off-by: Davidlohr Bueso <dbueso@suse.de>

---
 arch/s390/kernel/vdso.c | 2 --
 1 file changed, 2 deletions(-)

diff --git a/arch/s390/kernel/vdso.c b/arch/s390/kernel/vdso.c
index 4f07eaced960..604f4a68b524 100644
--- a/arch/s390/kernel/vdso.c
+++ b/arch/s390/kernel/vdso.c
@@ -299,8 +299,6 @@ static int __init vdso_init(void)
 
 	get_page(virt_to_page(vdso_data));
 
-	smp_mb();
-
 	return 0;
 }
 early_initcall(vdso_init);
-- 
2.6.6

